Mithril update 2023-10-11 (#268)
input-output-hk/cardano-updates: blog/2023-10-11-mithril.md

---
title: Mithril Team Update
slug: 2023-10-11-mithril
authors: jpraynaud
tags: [mithril]
hide_table_of_contents: false
---

## High level overview

> This week, the Mithril team has worked on the distribution of the Mithril client as a library that will be distributed on [crates.io](https://crates.io/), the Rust dependencies repository. The team has also worked on computing deterministically the transactions history of an address from the immutable files. Additionally, they have started to fix a bottleneck in the aggregator that limits the ingestion of signatures.

Reviewer: ```suggestion
This week, the Mithril team worked on the distribution of the Mithril client as a library that will be distributed on [crates.io](https://crates.io/), the Rust dependencies repository. The team also worked on computing the deterministic transaction history of an address from the immutable files and started fixing a bottleneck in the aggregator that limits the ingestion of signatures.
```

Reviewer: I am a bit confused about what 'deterministic' relates to:
- The team worked on deterministic computing of... 
- OR: The team also worked on computing the deterministic transaction history?

Author: The first option is correct: the computation is deterministic.
Does this work better?

```suggestion
This week, the Mithril team worked on the distribution of the Mithril client as a library that will be distributed on [crates.io](https://crates.io/), the Rust dependencies repository. The team also worked on the deterministic computation of the transaction history of an address from the immutable files and started fixing a bottleneck in the aggregator that limits the ingestion of signatures.
```

Reviewer: Yes, thanks for clarifying!

> Finally, they have kept working on the activation of Cloudflare on the infrastructure, and fixed some bugs in the nix CI.

Reviewer: ```suggestion
Finally, they kept working on the activation of Cloudflare on the infrastructure and fixed some bugs in the Nix CI.
```
